pkg/kubelet/prober/results/results_manager.go

```diff
@@ -87,3 +100,21 @@ func (m *manager) Remove(id kubecontainer.ContainerID) {
 	defer m.Unlock()
 	delete(m.cache, id)
 }
+
+func (m *manager) Updates() <-chan Update {
+	if m.updates == nil {
+		m.updates = make(chan Update)
+	}
+	return m.updates
+}
+
+// pushUpdates sends (non-blocking) an update on the updates channel if it is initialized.
+func (m *manager) pushUpdate(update Update) {
+	if m.updates != nil {
+		select {
+		case m.updates <- update:
+		default:
+			// Don't block
```

Comment:
Ack. I think the performance optimization (avoid syncing all pods) is necessary as we scale.